compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -1045,6 +1066,13 @@ LLVMRustOptimizeWithNewPassManager(
     }
   }
 
+  if (ExtraPassesLen) {
+    if (auto Err = PB.parsePassPipeline(MPM, StringRef(ExtraPasses, ExtraPassesLen))) {
+      std::string ErrMsg = toString(std::move(Err));
+      LLVMRustSetLastError(ErrMsg.c_str());
```

Comment:
Should return early here.

Reply:
I initially only emitted a warning, but on further consideration it makes sense to error and abort if invalid passes are specified. This required a few more changes to propagate the error.